opentelemetry-otlp/tests/integration_test/tests/integration_tests.rs

```diff
@@ -0,0 +1,95 @@
+use integration_test_runner::images::Collector;
+use std::fs::File;
+use std::os::unix::fs::PermissionsExt;
+use std::time::Duration;
+use testcontainers::clients::Cli;
+use testcontainers::core::Port;
+use testcontainers::RunnableImage;
+
+mod traces;
+
+const COLLECTOR_CONTAINER_NAME: &str = "otel-collector";
+const TEST_RESULT_DIR_IN_CONTAINER: &str = "testresults";
+const EXPECTED_DIR: &str = "./expected";
+
+struct TestSuite {
+    result_file_path: &'static str,
+}
+
+impl TestSuite {
+    fn new(result_file_path: &'static str) -> Self {
+        Self { result_file_path }
+    }
+
+    pub fn expected_file_path(&self) -> String {
+        format!("{}/{}", EXPECTED_DIR, self.result_file_path)
+    }
+
+    pub fn result_file_path_in_container(&self) -> String {
+        format!(
+            "/{}/{}",
+            TEST_RESULT_DIR_IN_CONTAINER, self.result_file_path
+        )
+    }
+
+    pub fn result_file_path(&self) -> String {
+        format!("./{}", self.result_file_path)
+    }
+
+    /// Create a empty file on localhost and copy it to container with proper permissions
+    /// we have to create the file for the container otherwise we will encounter a permission denied error.
+    /// see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3159
+    pub fn create_temporary_result_file(&self) -> File {
+        let file = File::create(self.result_file_path()).unwrap();
+        file.set_permissions(std::fs::Permissions::from_mode(0o666))
+            .unwrap();
+        file
+    }
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+#[ignore] // skip when running unit test
+async fn integration_tests() {
+    let test_suites = [TestSuite::new("traces.json")];
+
+    let mut collector_image = Collector::default();
+    for test in test_suites.as_ref() {
+        let _ = test.create_temporary_result_file();
+        collector_image = collector_image.with_volume(
+            test.result_file_path().as_str(),
+            test.result_file_path_in_container().as_str(),
+        );
+    }
+
+    let docker = Cli::default();
+    let mut image =
+        RunnableImage::from(collector_image).with_container_name(COLLECTOR_CONTAINER_NAME);
+
+    for port in [
+        4317,  // gRPC port
+        4318,  // HTTP port
+        55681, // OpenTelemetry protocol port
+        8888,  // Prometheus metrics exposed by the collector
```

Comment:
do we need this? I think only 4317, 4318 should be needed for testing the gRPC and Http based exporting and nothing more.